app/src/main/java/com/example/mobile_app_hw1/MainActivity.kt

```diff
@@ -1,11 +1,42 @@
 package com.example.mobile_app_hw1
 
 import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.DividerItemDecoration
 
-class MainActivity:AppCompatActivity() {
+class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        val recyclerView: RecyclerView = findViewById(R.id.recycler_view)
+        recyclerView.layoutManager = LinearLayoutManager(this)
+
+        val divider = DividerItemDecoration(this, DividerItemDecoration.VERTICAL)
+        recyclerView.addItemDecoration(divider)
+
+        val recipes = listOf(
+            Recipe(1, "STEAK", imageUrl = R.drawable.steak),
+            Recipe(2, "KHINKALI", imageUrl = R.drawable.khinkali),
+            Recipe(3, "KHACHAPURI", imageUrl = R.drawable.khachapuri)
+        )
+
+        val adapter = RecipeAdapter(
+            recipes,
+            itemClickListener = { recipe ->
+                Toast.makeText(this, "Clicked: ${recipe.title}", Toast.LENGTH_SHORT).show()
```

Comment:
Instead of showing the recipe's title when an item is clicked, it might be more useful to display the id of the recipe